kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm-search/src/test/java/org/kie/server/integrationtests/jbpm/search/TaskSearchServiceIntegrationTest.java

```diff
@@ -0,0 +1,304 @@
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK);
+        Assertions.assertThat(processInstanceId).isNotNull();
+
+        List<TaskSummary> tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+        Assertions.assertThat(tasks.size()).isGreaterThan(0);
+        TaskSummary task = tasks.get(0);
+        testFindTaskInstanceWithSearchService(createQueryFilterGreaterThan(TaskField.ACTIVATIONTIME, Date.from(Instant.EPOCH)), task.getId());
+    }
+
+    @Test
+    public void testFindTaskWithParentIdGreaterThanAndEqualsToFilter() throws Exception {
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK);
+        Assertions.assertThat(processInstanceId).isNotNull();
+
+        List<TaskSummary> tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+        Assertions.assertThat(tasks.size()).isGreaterThan(0);
+        TaskSummary task = tasks.get(0);
+        testFindTaskInstanceWithSearchService(createQueryFilterGreaterThanAndEqualsTo(TaskField.PARENTID, -2, -1), task.getId());
+    }
+
+    @Test
+    public void testFindTaskWithAndEqualsToFilter() throws Exception {
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK);
+        Assertions.assertThat(processInstanceId).isNotNull();
+
+        List<TaskSummary> tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+        Assertions.assertThat(tasks.size()).isGreaterThan(0);
+        TaskSummary task = tasks.get(0);
+
+        HashMap<TaskField, Comparable<?>> compareList = new HashMap<>();
+        compareList.put(TaskField.TASKID, task.getId());
+        compareList.put(TaskField.DEPLOYMENTID, CONTAINER_ID);
+        compareList.put(TaskField.PROCESSINSTANCEID, processInstanceId);
+        compareList.put(TaskField.NAME, "First task");
+        compareList.put(TaskField.CREATEDBY, USER_YODA);
+        compareList.put(TaskField.ACTUALOWNER, USER_YODA);
+        compareList.put(TaskField.DESCRIPTION, "");
+        compareList.put(TaskField.DUEDATE, task.getExpirationTime());
+        compareList.put(TaskField.PRIORITY, task.getPriority());
+        compareList.put(TaskField.STATUS, task.getStatus());
+
+        List<Long> resultsIds = new ArrayList<>();
+        List<TaskInstance> results = null;
+
+        results = searchServicesClient.
+                findHumanTasksWithFilters(createQueryFilterAndEqualsTo(compareList), 0, 100);
+
+        resultsIds = new ArrayList<>();
+        for (TaskInstance res : results) {
+            resultsIds.add(res.getId());
+        }
+
+        Assertions.assertThat(results).isNotNull();
+        Assertions.assertThat(results.size()).isGreaterThanOrEqualTo(0);
+        Assertions.assertThat(resultsIds).contains(task.getId());
+
+        final TaskInstance[] instance = new TaskInstance[1];
+        results.forEach((t) -> {
```

Comment:
The same as above; there better ways of retrieving a particular item from a collection.